### ISSUE TITLE
How to allow to write a filtration condition in one line with WHERE keyword

### ISSUE THREAD

**Issue by Kirill-Ulich** [sqlfluff/sqlfluff#6460]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Issue/Suggested Improvement

I use a construction of this type in many places:
```sql
WHERE 1 = 1
    AND col1 = 'a'
    AND ...
```
This is convenient because I can comment out a line or several lines when debugging and nothing will break.
However, sqlfluff requires moving `1 = 1` to a new line. Is there any way I can configure sqlfluff to allow such constructions?
`Noqa` is not suitable because such a construction is very often used. Ignoring the LT02 rule is also not suitable because other indents issues should not be ignored.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by Kirill-Ulich** [comment]
I tried using `allow_implicit_indents`. For `where 1 = 1` it helped, but now the linter started complaining about other indents. So another solution is needed.

**Comment by peterbud** [comment]
It's an old issue, and in the meantime config has been renamed/improved.
In SQLFluff, this behavior is controlled by indentation semantics, and the only built-in switch is:

```ini
[sqlfluff:indentation]
implicit_indents = allow
```

With this enabled, SQLFluff can accept both styles:
* `WHERE 1 = 1 ...`
* `WHERE newline + indented conditions`

So if your goal is “allow this pattern but still keep other indent checks”, this is the closest supported option.
There is currently no fine-grained config like “allow implicit indent only when first condition is 1 = 1”.

If you want strict behavior elsewhere, the alternatives are:
* keep `implicit_indents = allow` and use targeted `-- noqa: LT02` (you said this is too frequent), or
* create a custom rule/plugin for your team style.
